base/database.py

```diff
@@ -4025,7 +4033,9 @@ def iterate(cls):
     @classmethod
     def length(cls):
         '''Return the number of marks in the database.'''
-        return len(builtins.list(cls.iterate()))
+        iterable = cls.iterate()
+        listable = [item for item in iterable]
```

Comment:
This doesn't need an intermediary `iterable`, and can just be a straight-up list-comprehension.